### ISSUE TITLE
Add string / and /= operators

### ISSUE THREAD

**Issue by stryku** [stryku/cmakesl#87]
```cpp
auto foo = "bar" / "baz" // foo == "bar/baz"
foo /= "qux" // foo == "bar/baz/qux"
```

**Comment by DoYouEvenCpp** [comment]
Isn't `operator/` and `operator=/` a bit misleading (not needed?) for the string type?
I'd expect the end-user to use:
```
auto foo = "bar";
foo += "/";
foo += "bar";
```
and so on.

**Comment by stryku** [comment]
String type is the type used to deal with paths. In CMake there is a lot of path operations (a lot of concatenations bother us in this case). I believe that these operators would help to deal with path concatenations, so the code would be clearer.

Additioanlly, `boost::filesystem::path` and `std::filesystem::path` provide these operators.

**Comment by DoYouEvenCpp** [comment]
Got your point.

Then maybe there should be used not an operator but a custom method, i.e.: `string concatPath(string value, string delimiter = "/");` ?
Delimiter might not event be needed in that case.

**Comment by stryku** [comment]
Fixed in #227
